util/module_logger.go

```diff
@@ -0,0 +1,260 @@
+package util
```

Comment:
It feels to me like the module logger should be in its own package. This would help us keep strong separation of concerns, particularly if we want to move or release the module logger later as a standalone tool.